<a href="https://colab.research.google.com/github/drewjhowells/CSE450/blob/main/Assessment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Introduction
This assignment will test how well you're able to perform various data science-related tasks.

Each Problem Group below will center around a particular dataset that you have worked with before.

To ensure you receive full credit for a question, make sure you demonstrate the appropriate pandas, altair, or other commands as requested in the provided code blocks.

You may find that some questions require multiple steps to fully answer. Others require some mental arithmetic in addition to pandas commands. Use your best judgment.

## Submission
Each problem group asks a series of questions. This assignment consists of two submissions:

1. After completing the questions below, open the Module 01 Assessment Quiz in Canvas and enter your answers to these questions there.

2. After completing and submitting the quiz, save this Colab notebook as a GitHub Gist (You'll need to create a GitHub account for this), by selecting `Save a copy as a GitHub Gist` from the `File` menu above.

    In Canvas, open the Module 01 Assessment GitHub Gist assignment and paste the GitHub Gist URL for this notebook. Then submit that assignment.

## Problem Group 1

For the questions in this group, you'll work with the Netflix Movies Dataset found at this url: [https://raw.githubusercontent.com/byui-cse/cse450-course/master/data/netflix_titles.csv](https://raw.githubusercontent.com/byui-cse/cse450-course/master/data/netflix_titles.csv)


### Question 1
Load the dataset into a Pandas data frame and determine what data type is used to store the `release_year` feature.

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("https://raw.githubusercontent.com/byui-cse/cse450-course/master/data/netflix_titles.csv")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6234 entries, 0 to 6233
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       6234 non-null   int64 
 1   type          6234 non-null   object
 2   title         6234 non-null   object
 3   director      4265 non-null   object
 4   cast          5664 non-null   object
 5   country       5758 non-null   object
 6   date_added    6223 non-null   object
 7   release_year  6234 non-null   int64 
 8   rating        6224 non-null   object
 9   duration      6234 non-null   object
 10  listed_in     6234 non-null   object
 11  description   6234 non-null   object
dtypes: int64(2), object(10)
memory usage: 584.6+ KB


1) **release_year data type:** int64

### Question 2
Filter your dataset so it contains only `TV Shows`. How many of those TV Shows were rated `TV-Y7`?

In [ ]:
df.head()
tv_df = df[df['type'] == ("TV Show")]
tv_df = tv_df[tv_df['rating'] == "TV-Y7"]
tv_df.count()

show_id         100
type            100
title           100
director          7
cast             93
country          77
date_added       99
release_year    100
rating          100
duration        100
listed_in       100
description     100
dtype: int64

2) TV-Y7 rated Shows: 100

### Question 3
Further filter your dataset so it only contains TV Shows released between the years 2000 and 2009 inclusive. How many of *those* shows were rated `TV-Y7`?

In [ ]:
q3_df = tv_df.query('release_year >= 2000').query('release_year <= 2009')
q3_df

show_id     type                   title director  \
4528  70213120  TV Show  Beyblade: Metal Fusion      NaN   
5836  70177034  TV Show               Yu-Gi-Oh!      NaN   
6098  70308794  TV Show  What's New Scooby-Doo?      NaN   
6117  60036794  TV Show                 Sonic X      NaN   

                                                   cast        country  \
4528  Aki Kanada, Satoshi Hino, Emiri Kato, Kei Shin...          Japan   
5836  Dan Green, Eric Stuart, Amy Birnbaum, Darren D...          Japan   
6098  Frank Welker, Casey Kasem, Mindy Cohn, Grey De...  United States   
6117  Suzanne Goldish, Dan Green, Jason Anthony Grif...          Japan   

              date_added  release_year rating   duration  \
4528   December 15, 2017          2009  TV-Y7   1 Season   
5836      March 31, 2018          2005  TV-Y7  5 Seasons   
6098   December 15, 2019          2005  TV-Y7  3 Seasons   
6117    December 1, 2019          2004  TV-Y7  4 Seasons   

                   listed_in  \
4528  Anime Series, Kids' TV   
5836  Anime Series, Kids' TV   
6098   Kids' TV, TV Comedies   
6117  Anime Series, Kids' TV   

                                            description  
4528  In this anime adventure series for kids, the f...  
5836  The lives of young Yugi Moto and his friends J...  
6098  Scooby-Doo and the gang enter the 21st century...  
6117  While battling the evil Dr. Eggman, Sonic gets...

3) 2000-2009 TV-Y7: 4

## Problem Group 2

For the questions in this group, you'll work with the Cereal Dataset found at this url: [https://raw.githubusercontent.com/byui-cse/cse450-course/master/data/cereal.csv](https://raw.githubusercontent.com/byui-cse/cse450-course/master/data/cereal.csv)


### Question 4
After importing the dataset into a pandas data frame, determine the median amount of `protein` in cereal brands manufactured by Kelloggs. (`mfr` code "K")

In [ ]:
cereal_df = pd.read_csv("https://raw.githubusercontent.com/byui-cse/cse450-course/master/data/cereal.csv")
cereal_df.head()

kelloggs = cereal_df[cereal_df['mfr']== 'K']
median = kelloggs['protein'].median()
median

3.0

4) Median Kelloggs Protein: 3.0g

### Question 5
In order to comply with new government regulations, all cereals must now come with a "Healthiness" rating. This rating is calculated based on this formula:

    healthiness = (protein + fiber) / sugar

Create a new `healthiness` column populated with values based on the above formula.

Then, determine the median healthiness value for only General Mills cereals (`mfr` = "G"), rounded to two decimal places.

In [ ]:
cereal_df['healthiness'] = (cereal_df['protein'] + cereal_df['fiber'])/cereal_df['sugars']
gm_df = cereal_df[cereal_df['mfr'] == 'G']
gm_df['healthiness'].median()

0.475

5) Median GM Healthiness: 0.475 -> 0.48

## Problem Group 3

For the questions in this group, you'll work with the Titanic Dataset found at this url: [https://raw.githubusercontent.com/byui-cse/cse450-course/master/data/titanic.csv](https://raw.githubusercontent.com/byui-cse/cse450-course/master/data/titanic.csv)

### Question 6

After loading the dataset into a pandas DataFrame, create a new column called `NameGroup` that contains the first letter of the passenger's surname in lower case.

Note that in the dataset, passenger's names are provided in the `Name` column and are listed as:

    Surname, Given names

For example, if a passenger's `Name` is `Braund, Mr. Owen Harris`, the `NameGroup` column should contain the value `b`.

Then count how many passengers have a `NameGroup` value of `k`.

In [ ]:
titanic = pd.read_csv("https://raw.githubusercontent.com/byui-cse/cse450-course/master/data/titanic.csv")
titanic.head()

PassengerId Survived  Pclass  \
0            1       No       3   
1            2      Yes       1   
2            3      Yes       3   
3            4      Yes       1   
4            5       No       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [ ]:
titanic['NameGroup'] = titanic['Name'].str.slice(0, 1).str.lower()
k_titanic = titanic[titanic['NameGroup'] == 'k']
k_titanic.count()

PassengerId    28
Survived       28
Pclass         28
Name           28
Sex            28
Age            16
SibSp          28
Parch          28
Ticket         28
Fare           28
Cabin           5
Embarked       28
NameGroup      28
dtype: int64

6) 'k' Name Group Count: 28